### PR TITLE
AMDGPU blog

### PR DESCRIPTION
**Pull request by jcfrosty**
this is my first time attempting to add to the blog. Can someone please double check my work?

### CODE REVIEW

File: content/article/AMDGPU_20170624.md [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+Differences in AMDGPU and AMDGPU-PRO, once complete, will be minimal. 
+AMDGPU-PRO is a "hybrid" driver meaning paritially open and closed source.
+Nearly everything in AMDGPU-PRO will be in AMDGPU open source. AMDGPU-PRO
+will be geared for supporting FirePRO cards in workstations and tested
+to support things such as solidworks and other business software. 
+
+Currently, there are some gaps in support. Where PRO supports Freesync, HDMI 2.0,
+HDMI Audio, OpenCL, etc., the AMDGPU driver will eventually attain all that
+through what is known as Display Code (DC) or DAL/HAL (Display/Hardware abstraction
+layer).  The old DAL was just as it sounds, an abastraction layer. This
```
Comment (Ryuno-Ki) [inline content/article/AMDGPU_20170624.md:32]: There are two spaces after the period of the sentence concerning DAL, HAL.

File: content/article/AMDGPU_20170624.md [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+to be native or direct communication with the hardware. This is not allowed in
+the Linux Kernel and for good reason. We would want native and direct communication
+with hardware which is better for performance and being maintained and clean.
+
+So why did AMD use a HAL? Well, the simplest reason was for sharing code.
+AMD's software driver team is small. To save time, money, and effort, using
+a HAL was a simple method. They could write code that (mostly) worked for
+both Windows and Linux at the same time. So primary focus on Windows translated
+to some focus on Linux. Unfortunately, this doesn't mean it worked well.
+We all remember how 'lovely' fglrx was a times with broken crossfire support,
```
Comment (Ryuno-Ki) [inline content/article/AMDGPU_20170624.md:43]: Could you please use proper quotation marks? Like `„`, `“` or `”`?

File: content/article/AMDGPU_20170624.md [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+broken eyefinity setup, poor kernel support, etc.
+
+OK so small software dev team, HAL isn't fully working in this instance,
+what is the solution? They decided to go open source. "Great, shouldn't this
+be simple?" Short answer, No. They have to go over code with a fine tooth comb
```
Comment (Ryuno-Ki) [inline content/article/AMDGPU_20170624.md:48]: Again, please use proper quotation marks: `„`, `“` and `”`.

File: content/article/AMDGPU_20170624.md [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+to make sure they aren't releasing code they "don't own" persay. Can't exactly
+opensource DirectX related etc. After that, they also have to remove the HAL and recreate
+code from scratch to make a native/direct communication line for the linux kernel.
```
Comment (Ryuno-Ki) [inline content/article/AMDGPU_20170624.md:51]: Here you used a lowercase `Linux`. On purpose?

File: content/article/AMDGPU_20170624.md [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+no freesync, no HDMI 2.0, no VEGA and beyond, Vulkan support may be hit
+and miss. If you're using an HD7XXX or R9 2XX card, your drivers may change
+after a kernel upgrade resulting in different behavior. People looking
+for AMDGPU-PRO should and probably would abandon that thought considering
+the performance gaming-wise is better off on the open source driver.
+You may also have noticed the bump in MESA lately to 17.1? MESA may get
+bumped more frequently due to the constant development for AMDGPU.
+
+We understand you may feel some frustration in this time awaiting fixes,
+but let it be known this issue is affecting all Linux distributions.
```
Comment (Ryuno-Ki) [inline content/article/AMDGPU_20170624.md:102]: Hm, I would need to think about how to interpret the phrase „let it be known” …

File: content/article/AMDGPU_20170624.md [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+Its the typical growing pains of Linux when a large change takes place.
+We are keeping on top of it and trying to stay up-to-date bringing fixes,
+and performance boosts, but maintaining stability. 
+
+We hope you're enjoying Sabayon just as much as we do!
+-Frosty
```
Comment (Ryuno-Ki) [inline content/article/AMDGPU_20170624.md:108]: That line will be placed after the one above. Are you fine with that?

**Review by Ryuno-Ki — CHANGES_REQUESTED** [review]
Thanks for taking time to write this article, @jcfrosty!

I found some minor bits (Most the times I commented them once, but they explain to further occurences in this text).

I request changes mainly because I would like to see more level-two-headlines (`##`), in order to help the reader grasp the content easier. As an example read only the headlines and tell me, what the content is about.

Since the topic is quite technical, you should add link to a wiki (ours or Wikipedia) for technical terms like fgrlx, DC or similiar (my personal opinion).

If you add the headlines, I approve the PR. The other changes just ease reading IMHO.

**Comment by Ryuno-Ki** [comment]
Hi, @jcfrosty!

Could you tell me, why you closed this PR?

**Comment by jcfrosty** [comment]
I didn't? Or at least I didn't mean to? I'm trying to correct the problems
you listed currently.

File: content/article/AMDGPU_20170624.md [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+opensource DirectX related etc. After that, they also have to remove the HAL and recreate
+code from scratch to make a native/direct communication line for the linux kernel.
+You could say this is essentially rewriting the driver from scratch. Its also
+more than just a display driver. You have Vulkan, OpenCL, HSA, VCE, UVD and the many 
+other interfaces in which one may use the graphics card. 
+
+So when can we expect to see support for all items the display code brings us?
+Thats an excellent, but daunting question. 
+AMD is about to meet the crossroads in which they either: 
+* Do not support their latest graphics cards at all ,VEGA and beyond, upon release
```
Comment (Enlik) [inline content/article/AMDGPU_20170624.md:59]: " ,VEGA" → ", VEGA"

File: content/article/AMDGPU_20170624.md [hunk 8]
```diff
@@ -0,0 +1,110 @@
...
+gotten its hands on AMDGPU, its outperforming AMDGPU-PRO in several instances.
+Is the opensource driver perfect? No, far from it, but its growing by leaps
+and bounds at an accelerated rate. Even Valve has dedicated some resources
+to assisting in the growth of AMDGPU.
+
+What about Vulkan support? Its there. AMDGPU has radv which is also growing
+quickly. radv is the radeon opensource vulkan project. Have you tried DOTA2
+with vulkan lately? It works. [See the benchmarks and gaming results here.](https://www.phoronix.com/scan.php?page=article&item=amdgpu-new-1710&num=1)
+
+Valve has also offered is services to help forward AMDGPU. [Looking forward to VR.](https://www.phoronix.com/scan.php?page=news_item&px=AMDGPU-PRO-VR-Linux)
```
Comment (Enlik) [inline content/article/AMDGPU_20170624.md:86]: is → its

Comment (Enlik) [inline content/article/AMDGPU_20170624.md:86]: Try to make a note what VR is.

File: content/article/AMDGPU_20170624.md [hunk 9]
```diff
@@ -0,0 +1,110 @@
...
+code from scratch to make a native/direct communication line for the linux kernel.
+You could say this is essentially rewriting the driver from scratch. Its also
+more than just a display driver. You have Vulkan, OpenCL, HSA, VCE, UVD and the many 
+other interfaces in which one may use the graphics card. 
+
+So when can we expect to see support for all items the display code brings us?
+Thats an excellent, but daunting question. 
+AMD is about to meet the crossroads in which they either: 
+* Do not support their latest graphics cards at all ,VEGA and beyond, upon release
+* Push their team to get the DC conversion completed and submitted to the linux kernel
```
Comment (Enlik) [inline content/article/AMDGPU_20170624.md:60]: It could be worth to "unwind" the acronym here.

File: content/article/AMDGPU_20170624.md (hunk 1 above)
Comment (jcfrosty) [reply content/article/AMDGPU_20170624.md:32]: fixed

File: content/article/AMDGPU_20170624.md (hunk 2 above)
Comment (jcfrosty) [reply content/article/AMDGPU_20170624.md:43]: fixed

File: content/article/AMDGPU_20170624.md (hunk 3 above)
Comment (jcfrosty) [reply content/article/AMDGPU_20170624.md:48]: fixed

File: content/article/AMDGPU_20170624.md (hunk 4 above)
Comment (jcfrosty) [reply content/article/AMDGPU_20170624.md:51]: fixed

File: content/article/AMDGPU_20170624.md (hunk 5 above)
Comment (jcfrosty) [reply content/article/AMDGPU_20170624.md:102]: fixed

File: content/article/AMDGPU_20170624.md (hunk 6 above)
Comment (jcfrosty) [reply content/article/AMDGPU_20170624.md:108]: fixed

File: content/article/AMDGPU_20170624.md (hunk 7 above)
Comment (jcfrosty) [reply content/article/AMDGPU_20170624.md:59]: fixed

File: content/article/AMDGPU_20170624.md (hunk 8 above)
Comment (jcfrosty) [reply content/article/AMDGPU_20170624.md:86]: fixed

Comment (jcfrosty) [reply content/article/AMDGPU_20170624.md:86]: fixed

File: content/article/AMDGPU_20170624.md (hunk 9 above)
Comment (jcfrosty) [reply content/article/AMDGPU_20170624.md:60]: Fixed

**Comment by jcfrosty** [comment]
Sorry for the mess. as you can tell, I'm not completely familiar with github yet. I'm learning ;) 

https://github.com/Sabayon/sabayon-website-next/pull/55

**Comment by Enlik** [comment]
General comment: I enjoyed the read but my feel is that the article is more technical (due to vocabulary and style being used), I could said nerdy, and can be harder by the less technical ones, or can leave readers with less knowledge of the video/GPU ecosystem some questions.

By itself, this is not a bad thing, especially given that it's signed by your name (meaning the style /can/ be personal as I think of it), and in my option it could be left as is, but you might keep this in mind when writing futher articles on the website. (I'm not for dumbing down content; reading something like this actually allows someone to learn from it which is great, so take what I wrote with a grain of salt.)

**Comment by Enlik** [comment]
> Sorry for the mess. as you can tell, I'm not completely familiar with github yet. I'm learning ;)

This one could be reopened instead, to keep everything in one place. Please do this like this in the future.

File: content/article/AMDGPU_20170624.md (hunk 9 above)
Comment (Enlik) [reply content/article/AMDGPU_20170624.md:60]: "Push their team to get the display code converted and submitted to the Linux kernel"

Better but now it reads like a generic display code (so why would it need converting?), not *that* DC stat stands next to HAL and DAL.

**Comment by jcfrosty** [comment]
Glad you liked it. Yeah, I figured the only people who would be interested in the AMDGPU blog would be people with an AMD card looking for answers (both techy and non). I don't plan on them all being techy, but there are those who are upset with AMD in general and I felt they deserved some sort of explanation. Even if it was a little over the top, they could still get something out of it.

I'll keep in mind the ability to reopen next time. I was looking at the closed image at the top and looking to see if I could click it and get it to reopen, but it doesn't work that way. I just noticed the "reopen and comment" button on the bottom. I should have checked that before opening a new one.

**Comment by Enlik** [comment]
Changing "radeon" to something like "the radeon driver" should make it much easier for those non-AMD device owners in the following sentence.

"Some of you may still be using radeon on 
+AMD HD7XXX or R9 2XX (SI/CIK) video cards."

**Comment by jcfrosty** [comment]
Good point, also fixed.

**Pull request closed without merging**